contributors/devel/cherry-picks.md

```diff
@@ -7,39 +7,42 @@ depending on the point in the release cycle.
 ## Propose a Cherry Pick
 
 1. Cherrypicks are [managed with labels and milestones](/contributors/guide/release-notes.md)
-1. To get a PR merged to the release branch, first ensure the following labels
-   are on the original **master** branch PR:
-  * The milestone for the branch that you want to cherry-pick to (e.g. v1.7 if you want to backport your change in 1.7 branch)
-  * The `cherrypick-candidate` label
-1. If `release-note-none` is set on the master PR, the cherrypick PR will need
-   to set the same label to confirm that no release note is needed.
-1. `release-note` labeled PRs generate a release note using the PR title by
-   default OR the release-note block in the PR template if filled in.
-  * See the [PR template](https://git.k8s.io/kubernetes/.github/PULL_REQUEST_TEMPLATE.md) for more details.
-  * PR titles and body comments are mutable and can be modified at any time
-    prior to the release to reflect a release note friendly message.
+
```

Comment:
This is more of a note to self as I work through this:
- `submit-queue` is actually the thing responsible for merging PR's to `release-1.x`
- `submit-queue` won't bother if there is a `do-not-merge/cherry-pick-unapproved` label
- the `label-pick-unapproved` munger run as part of the `misc-mungers` deployment applies this label
  - it removes the label if there is a `cherrypick-approved` label
  - it applies the label if a PR is not for `master`
- the `cherrypick-approved` label can either be applied by a human, or by the `cherrypick-auto-approve` munger
- the `cherrypick-auto-approve` munger
  - only looks at PR's not for `master`
  - skips PR's that have `cherrypick-approved` and a milestone
  - skips PR's that don't mention other PR's (hereafter called "parents") in their description
  - skips PR's that are for `release-1.x` but have milestone `v1.y`
  - skips PR's whose parents don't have the `cherrypick-approved` label
  - skips PR's that are for `release-1.x` whose parents have milestone `v1.y` or none
  - applies `cherrypick-approved` and milestone `1.x` if it's made it this far
- various mungers will remove the `cherrypick-candidate` label
  - from "parent" PR's that have successfully merged
  - from "parent" PR's that lack a milestone
  - etc.

Reply:
Some thoughts:
- the current auto-approve process assumes one PR to master == one cherrypick, and discounts that some fixes may need to be backported to multiple release branches
- I had to double-check, but it sure looks like `cherrypick-candidate` is just used for display purposes, I'm not sure if there is automation that consumes the cherrypick queue
- Given that there is no bot-driven way to apply the `cherrypick-candidate` label, it's more likely a PR author without repo write access would just run the cherypick script themselves

I dislike this whole process and suggest that it's _way_ easier to just get a human to apply the `cherrypick-approved` label

Reply:
Re: multi-branch pick tracking/process, see suggestions in https://groups.google.com/forum/m/#!topic/kubernetes-wg-contribex/XxYn02YQcV0

Reply:
thanks for reminding me of that thread, replied over there